pkg/git/gogit/transport.go

```diff
@@ -104,5 +106,13 @@ func (s *PublicKeyAuth) Method(secret corev1.Secret) (*git.Auth, error) {
 	}
 	pk.HostKeyCallback = callback
 
+	// override default ssh protocol to use our preferred key algorithms
+	cfg, err := pk.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	pkgssh.SetPreferredKeyAlgos(cfg)
+	transportClient.InstallProtocol("ssh", ssh.NewClient(cfg))
```

Comment:
Instead of doing this here, I would probably add it to an `init` function in this package so that it is executed on import and just once.

Reply:
Setting a global ssh config in the init returns an error (`ssh: must specify HostKeyCallback`). This is because the global config has a nil `HostKeyCallback` and overrides the one from the auth method that has the error:
See https://github.com/go-git/go-git/blob/e5bbc4d10904554dbc8fd5afd06f66814f7b173e/plumbing/transport/ssh/common.go#L120-L127